ghost/core/core/server/services/members/members-api/repositories/MemberRepository.js

```diff
@@ -1036,22 +1038,26 @@ module.exports = class MemberRepository {
             logging.error(e);
         }
 
-        let stripeCouponId = stripeSubscriptionData.discount && stripeSubscriptionData.discount.coupon ? stripeSubscriptionData.discount.coupon.id : null;
+        let stripeCouponId = stripeSubscriptionData.discount?.coupon?.id;
 
         // For trial offers, offer id is passed from metadata as there is no stripe coupon
         let offerId = data.offerId || null;
-        let offer = null;
 
-        if (stripeCouponId) {
-            // Get the offer from our database
-            offer = await this._offerRepository.getByStripeCouponId(stripeCouponId, {transacting: options.transacting});
-            if (offer) {
-                offerId = offer.id;
-            } else {
-                logging.error(`Received an unknown stripe coupon id (${stripeCouponId}) for subscription - ${stripeSubscriptionData.id}.`);
+        if (stripeCouponId && !offerId && ghostProduct) {
+            const cadence = _.get(subscriptionPriceData, 'recurring.interval', 'month');
```

Comment:
Why do we default to `'month'` here? Prices used for subscriptions must have a recurring property